lib/os_places_api/client.rb

```diff
@@ -18,10 +18,16 @@ def locations_for_postcode(postcode)
       validate_response_code(response)
 
       begin
-        json = JSON.parse(response)
+        json = JSON.parse(response.body)
         raise UnexpectedResponse if json["results"].nil?
 
-        json["results"]
+        json["results"].map do |result|
+          Location.new(postcode: result["DPA"]["POSTCODE"],
```

Comment:
Are we confident that the JSON response will always contain the fields we require. If there's a breaking change, and the structure changes we are quite tightly bound here. Maybe we could do something like

```
def build_location(result)
  return unless result["DPA"] # or raise an error
  address = result["DPA"]["address"]
  latitude = etc...
  Location.new(
    address: address,
    latitude: latitude,
    etc...
  )
end
```